packages/cozy-client/src/devtools/Pouch.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useMemo, useState, useCallback, useEffect } from 'react'
+import CozyPouchLink from 'cozy-pouch-link'
+import Button from 'cozy-ui/transpiled/react/MuiCozyTheme/Buttons'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import format from 'date-fns/format'
+
+import useClient from '../hooks/useClient'
+import PanelContent from './PanelContent'
+
+const listStyle = { paddingLeft: '1rem' }
+
+/**
+ * Allows to view state and manage the PouchLink of the current cozy
+ * client.
+ */
+const PouchDevTool = () => {
+  const client = useClient()
+  const [status, setStatus] = useState('NA')
+  const pouchLink = useMemo(() => {
+    return client.links.find(link => link instanceof CozyPouchLink)
+  }, [client])
+  const [resetting, setResetting] = useState(false)
+  useEffect(() => {
+    const handleReplicationStart = () => setStatus('replicating')
+    const handleReplicationStop = () => setStatus('stopped')
+    const handleReplicationEnd = () => setStatus('idle')
+    client.on('pouchlink:sync:start', handleReplicationStart)
+    client.on('pouchlink:sync:stop', handleReplicationStop)
+    client.on('pouchlink:sync:end', handleReplicationEnd)
+    return () => {
+      client.removeListener('pouchlink:sync:start', handleReplicationStart)
+      client.removeListener('pouchlink:sync:stop', handleReplicationStop)
+      client.removeListener('pouchlink:sync:end', handleReplicationEnd)
+    }
+  }, [client])
+
+  const handleClickStartReplication = useCallback(async () => {
+    // Need to stop the replication loop before restarting it
+    // otherwise startReplication has no effect since the replication
+    // loop is already started.
+    pouchLink.stopReplication()
+    pouchLink.startReplication()
+  }, [pouchLink])
+
+  const handleClickStopReplication = useCallback(async () => {
+    pouchLink.stopReplication()
+  }, [pouchLink])
+
+  const handleClickReset = useCallback(async () => {
+    setResetting(true)
+
+    try {
+      await pouchLink.reset()
+      if (window.confirm('Pouches have been destroyed, reload the page ?')) {
+        window.location.reload()
+      }
+    } catch (e) {
+      setResetting(false)
+    }
+  }, [setResetting, pouchLink])
+
+  if (!pouchLink) {
+    return (
+      <PanelContent>
+        <p>Pouch is not active</p>
+      </PanelContent>
+    )
+  }
+  return (
+    <PanelContent>
+      <Typography variant="subtitle1">Doctypes saved in Pouch</Typography>
+      <ul style={listStyle}>
+        {pouchLink.doctypes.map(doctype => (
+          <li key={doctype}>{doctype}</li>
```

Comment:
I think I am going to put "managed" in Pouch, it will be clearer that this does not mean that a doctype as been replicated yet.

> I'm wondering if we shouldn't add this "persisted information" here.

What kind ?

Reply:
I think this is fixed by the latest changes, information is now displayed per doctype, and this information is persisted to localStorage.